test/unit/node_shared_test.rb

```diff
@@ -20,6 +20,32 @@ class NodeSharedTest < ActiveSupport::TestCase
     assert_equal 5, html.scan('notes-grid-test').length
     assert html.scan('<td class="title">').length > 1
   end
+ 
+  test 'that NodeShared can be used to convert short codes like [button:foo:https://google.com] into tables which list buttons' do
+    before = "Here are some notes in a table: \n\n[button:Press me:/questions] \n\n[button:Cancel:https://google.com]\n\n`[button:Cancel:https://google.com]` This shouldn't get recognized because it's in ` ticks.\n\nMake sense?"
+    html = NodeShared.button(before)
+    assert html
+    assert_equal 1, html.scan('<a class="btn btn-primary inline-button-shortcode').length
+    assert_equal 1, html.scan('href=').length
+    assert_equal 5, html.scan('inline-button-shortcode').length
+    assert html.scan('<td class="title">').length > 1
```

Comment:
Oh! And here too, we should not look for a table `td` element; in fact we can probably just delete this line.